### ISSUE TITLE
no-changed-when: improve documentation and testing

### ISSUE THREAD

**Issue by Pant** [ansible-community/ansible-lint#1681]
##### Summary

When using the `command:` module as in the example bellow, ansible-lint gives error `Commands should not change things if nothing needs doing`

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible --version
ansible [core 2.11.3] 
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/pant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  ansible collection location = /home/pant/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.9.5 (default, May 24 2021, 12:50:35) [GCC 11.1.0]
  jinja version = 2.11.3
  libyaml = True

$ ansible-lint --version
ansible-lint 5.1.2 using ansible 2.11.3
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Linux Anonymous 5.12.13-arch1-1 - x86_64 GNU/Linux

##### STEPS TO REPRODUCE

Example (from https://docs.ansible.com/ansible/latest/collections/ansible/builtin/command_module.html):

```
- name: Safely use templated variable to run command. Always use the quote filter to avoid injection issues
  ansible.builtin.command: cat {{ myfile|quote }}
  register: myoutput
```

Problem message: 

```
no-changed-when: Commands should not change things if nothing needs doing
```

<!--- ```console (paste below)

```
-->
<!--- HINT: You can paste gist.github.com links for larger files -->

##### Desired Behaviour
<!--- Describe what you expected to happen when running the steps above -->

There should not be an problem running a command with the `register` option.

##### Actual Behaviour
<!--- Describe what actually happened. If possible run with extra verbosity (-vvvv) -->

Problem occurs when running a command with the `register` option.

---

Also there are cases where a command needs to be run but `creates` files are uknown. For example:

```
- name: Install composer dependencies
  command: composer install -d /var/www/attendize
  args: 
    creates: /var/www/attendize/????unknown_files????
```

Shouldn't commands like these (in both cases) be passed from ansible-lint without lint errors/warnings?



> Note 1: In example, directory `/var/www/attendize/` already exists and is not empty.

> Note 2: We are trying to refactor our ansible cockbook with 100% ansible-lint compatibility. I wonder if that is even possible though.

**Comment by konstruktoid** [comment]
> There should not be an problem running a command with the `register` option.

Makes sense IMHO, I'll write a PR in a couple of days unless someone says that `command` with the `register` option should fail.

> > Note 2: We are trying to refactor our ansible cockbook with 100% ansible-lint compatibility. I wonder if that is even possible though.

It is, all my playbooks and roles are 100% `ansible-lint` compatibility, but even `ansible-lint` contains bugs or rules that are open to discussion.

There is also most-often a workaround as well, e.g:

```yaml
- name: Safely use templated variable to run command. Always use the quote filter to avoid injection issues
  ansible.builtin.command: cat {{ myfile|quote }}
  register: myoutput
  changed_when: myoutput.rc != 0
  failed_when: myoutput.rc != 0
```

**Comment by Pant** [comment]
> It is, all my playbooks and roles are 100% `ansible-lint` compatibility, but even `ansible-lint` contains bugs or rules that are open to discussion.

That is very nice to hear. I will have this in mind in similar situations.

> There is also most-often a workaround as well, e.g:

Thank you for this suggestion. I find it very useful and highly appreciate it!

---

Looking forward to your PR :)
